pkg/controllers/apply_controller.go

```diff
@@ -263,20 +265,29 @@ func isManifestModified(observedGeneration int64, curObj, workObj *unstructured.
 	return false
 }
 
-// isSameUnstructuredMeta compares the metadata of two unstructured object.
-func isSameUnstructuredMeta(obj1, obj2 *unstructured.Unstructured) bool {
-	// we just care about label/annotation and owner references
-	if !equality.Semantic.DeepEqual(obj1.GetLabels(), obj2.GetLabels()) {
-		return false
-	}
-	if !equality.Semantic.DeepEqual(obj1.GetAnnotations(), obj2.GetAnnotations()) {
-		return false
-	}
-	if !equality.Semantic.DeepEqual(obj1.GetOwnerReferences(), obj2.GetOwnerReferences()) {
-		return false
+// Determines if differences between two unstructured.Unstructured objects
+// differ in ways that warrant the update (reapply) of the object.
+func isUpdateWarranted(obj1, obj2 *unstructured.Unstructured) bool {
+	return !isSpecHashAnnotationEqual(obj1, obj2)
+}
+
+// Determines if two annotations (SpecHash) on the two argued objects are equal.
+func isSpecHashAnnotationEqual(obj1 *unstructured.Unstructured, obj2 *unstructured.Unstructured) bool {
+	return obj1.GetAnnotations()[specHashAnnotation] == obj2.GetAnnotations()[specHashAnnotation]
+}
```

Comment:
since the `isSpec` function only used once, I wonder if it makes sense to combine them?

Reply:
Yes this works for now. Originally I had two functions that determined if the update was warranted, but it was determined that the owner check occurred before it entered this routine. I'll combine.